### PR TITLE
Add option to hide some system messages

### LINKED ISSUE #2959: Ability to remove joined / left channel text
I would love if you added an option in the administration panel where you can choose to display when a user joined/left a room. Atleast the users on my server find this distracting / unnecessary. 

I think this option would be great to have.

Thanks in advance /Cedemore

### PR DESCRIPTION
**Pull request by rodrigok**
@RocketChat/core 

Closes #2959

Add options:
- Hide "User Added" messages
- Hide "User Removed" messages
- Hide "User Join" messages
- Hide "User Leave" messages
  

captura de tela 2016-10-27 as 14 26 34

### CODE REVIEW

**Comment by lunitic** [comment]
@rodrigok Cheers for this !!

**Comment by Molle500** [comment]
<3 @rodrigok

**Comment by lunitic** [comment]
Backported it to 0.39 and tested it .. works 👍

**Comment by rodrigok** [comment]
@RocketChat/core Can someone review this?

File: packages/rocketchat-ui/views/app/room.coffee [hunk 1]
```diff
@@ -28,7 +28,28 @@ Template.room.helpers
 		return isSubscribed(this._id)
 
 	messagesHistory: ->
-		return ChatMessage.find { rid: this._id, t: { '$ne': 't' }  }, { sort: { ts: 1 } }
+		hideMessagesOfType = []
+		RocketChat.settings.collection.find({_id: /Message_HideType_.+/}).forEach (record) ->
```
Comment (engelgabriel) [inline packages/rocketchat-ui/views/app/room.coffee]: Is this part of the code being executed every time this method is called? Couldn't we cache this? Maybe on a session variable to be reactive?

Comment (rodrigok) [inline packages/rocketchat-ui/views/app/room.coffee]: I did some tests and this will execute when you open the room and just one time

**Pull request merged**